app/code/Magento/SalesRule/Model/Coupon/CaptchaConfigProvider.php

```diff
@@ -42,10 +32,11 @@ class CaptchaConfigProvider implements ConfigProviderInterface
      * @param Helper $captchaData
      * @param CustomerSession $session
      */
-    public function __construct(StoreManagerInterface $storeManager, Helper $captchaData, CustomerSession $session)
-    {
-        $this->storeManager = $storeManager;
-        $this->captchaData = $captchaData;
+    public function __construct(
+        private readonly StoreManagerInterface $storeManager,
+        private readonly Helper $captchaData,
+        CustomerSession $session
```

Comment:
Why not promote this property?
```suggestion
        private readonly CustomerSession $session,
```